### ISSUE TITLE
How to change vector tile data in mapbox?

### ISSUE THREAD

**Issue by xiaofsu** [mapbox/mapbox-gl-js#8244]
Use MapBox on the web.

When I first load the vector tile I want to make him reload other vector tiles is still using this layer id.(Can I have a layer request data again?)

And I see #3037
Use **addSource** and **removeSource**
Will flash on the screen. This is not good.(I don't quite understand how to solve the problem of flickering in #1989.)
And  show/hide them
Will cause the browser to be severely stuck because of the large amount of data
I need to render nearly 20 kinds of data, and they are all 15 days.

It looks like **geojson's setData** and **image's updateImage**

Unfortunately, I didn't find a way to change the data in the vector.

So, I want to ask How do I change the request path in the vector and have the MapBox request the resource again and render it to the map.

But I don't know how to implement it, how to get it done.

Or you have a better way, please help me. Thank you

Browser: Chrome 75.0.3770.142
System: windows10
Mapbox-gl-js: 1.1.1

**Comment by xiaofsu** [comment]
And I see mapbox/mapbox-gl-style-spec＃570
It’s too bad for me, isn’t it?

**Comment by joedjc** [comment]
@xiaofsu I'm looking for this as well. I'm hoping this will fix https://github.com/mapbox/mapbox-gl-js/pull/8048. Not sure of the status of it though

**Comment by xiaofsu** [comment]
> @xiaofsu I'm looking for this as well. I'm hoping this will fix #8048. Not sure of the status of it though

I think this method can solve my problem.
But it's a bad news to be unsure of when this method can be used.🤔🤔🤔

**Comment by jacknkandy** [comment]
@xiaofsu at the moment you can try something like this https://github.com/mapbox/mapbox-gl-js/issues/3709#issuecomment-265346656 however I am using this "hack" and am sometimes am getting random errors (see https://github.com/mapbox/mapbox-gl-js/issues/8233). Hopefully #8048 will be what we're after!!

**Comment by xiaofsu** [comment]
> @xiaofsu at the moment you can try something like this [#3709 (comment)](https://github.com/mapbox/mapbox-gl-js/issues/3709#issuecomment-265346656) however I am using this "hack" and am sometimes am getting random errors (see #8233). Hopefully #8048 will be what we're after!!

Currently in use  #3709 ,But the screen is still blinking, I don't think this is the reason for the network, because the browser has helped me with the cache. Hope #8048 can help us.😂😂

**Comment by xiaofsu** [comment]
![TestDemo](https://user-images.githubusercontent.com/46067472/62094987-78f83a00-b2b1-11e9-9a17-56843c437941.gif)
This is my recording of the demo I am implementing now, you can see the obvious flashing
![image](https://user-images.githubusercontent.com/46067472/62095063-d4c2c300-b2b1-11e9-91bb-69e6452d696a.png)
These data are also cached by the browser.

**Comment by xiaofsu** [comment]
I think I did the wrong direction at the beginning, I have been thinking about multiple mbtiles files, and then requesting different files to achieve the purpose of weather refresh. But I didn't think about an mbtiles file. There are multiple weather information in this file, and then filter to control what weather is displayed in each time period. Obviously an mbtiles file is correct this way.
![111](https://user-images.githubusercontent.com/46067472/62178632-5c750400-b37c-11e9-9a13-cf2e5c52b8eb.gif)

**Comment by xiaofsu** [comment]
> @xiaofsu I'm looking for this as well. I'm hoping this will fix #8048. Not sure of the status of it though

> @xiaofsu at the moment you can try something like this [#3709 (comment)](https://github.com/mapbox/mapbox-gl-js/issues/3709#issuecomment-265346656) however I am using this "hack" and am sometimes am getting random errors (see #8233). Hopefully #8048 will be what we're after!!

I am going to close this question because I have found the answer I want. I hope that you can also use this method to solve the current problem.
![image](https://user-images.githubusercontent.com/46067472/62178745-b675c980-b37c-11e9-8b10-9171715895b0.png)

**Comment by DXnima** [comment]
I accidentally came across this blog post:  [medium](https://medium.com/@carmeladi/dynamically-refreshing-a-vector-tiles-map-2f9f5b632bd2) It suggests that setting `cache-control` on the server side can achieve dynamic refreshing without causing flickering.

The Nginx configuration is as follows, You can modify the refresh time:
```
location /map/ {
        proxy_pass  http://tile-server/;
        add_header Cache-Control "max-age=5";
    }
```

All you need to do is to load the tile map service normally. Mapbox will request the latest vector tiles when the cache time expires.
